### ISSUE TITLE
TPM 2.0 Support

### ISSUE THREAD

**Issue by colek42** [in-toto/in-toto-golang#134]
The TPM can be used for attestation of environment data and signing operations.  Support for TPM can be mostly added with the go-attestation library.

**Comment by shibumi** [comment]
Hi Cole, sorry for the delay. I really like what you are doing. Go ahead :)

We should just make sure that in-toto-golang remains functional if the environment has no TPM support . Meaning: we should not depend on it.

**Comment by colek42** [comment]
Agree, I'll look for a TPM in the default location and try use it if we find one.  I'll probably have a couple of extra CLI args as well.  I am working on a lot of this outside of the in-toto core as a library.   https://github.com/testifysec/go-ima/
Few things I'm working on

1. Support for IMA log validation
2. Support for using TPM for the signing process
3. Record TPM Quote as metadata
4. Record TPM PCR Registers as metadata
5. Record Event Log as metadata
6. Optionally include IMA lof as metadata
